core/src/utilities/random.cc

```diff
@@ -746,4 +950,136 @@ namespace jiminy
+    void computePerlinHeightmap(PerlinProcessType perlinProcess,
+                                Eigen::Rotation2D<double> rot_mat,
+                                double offset,
+                                bool is2D,
+                                const Eigen::Vector2d & pos,
+                                double & height,
+                                Eigen::Ref<Eigen::Vector3d> normal)
+    {
+        if (is2D)
+        {
+            // Compute the height
+            height = perlinProcess(static_cast<float>(pos[0]), static_cast<float>(pos[1]));
+
+            // Compute gradient of the Perlin Process and retrieve the normal
+            const std::pair<double, double> gradient =
+                perlinProcess.gradient(static_cast<float>(pos[0]), static_cast<float>(pos[1]));
+
+            // Compute the inverse of the normal's Euclidean norm
+            const double normInv =
+                1.0 / std::sqrt(std::pow(gradient.first, 2) + std::pow(gradient.second, 2));
+
+            // Update normal vector
+            // step 1. compute normal in Perlin process reference frame:
+            normal << -gradient.first * normInv, -gradient.second * normInv, normInv;
+            // step 2. Rotate normal vector in world plane reference frame:
+            normal.head<2>() = rot_mat * normal.head<2>();
+        }
+        else
+        {
+            // Compute the unidirectional Perlin Process relative coordinate
+            Eigen::Vector2d posRel = (rot_mat.inverse() * pos).array();
+
+            // Compute the height
+            float delta = std::abs(static_cast<float>(posRel[0] + offset));
+            height = perlinProcess(delta);
+
+            // Compute gradient of the Perlin Process and retrieve the normal
+            const double slope = perlinProcess.gradient(delta);
+
+            // Compute the inverse of the normal's Euclidean norm
+            const double normInv = 1.0 / std::sqrt(1.0 + std::pow(slope, 2));
+
+            // Update normal vector
+            // step 1. compute normal in Perlin process reference frame:
+            // normal << -slope * normInv, 0.0, normInv;
+            // step 2. Rotate normal vector in world plane reference frame:
+            // normal.head<2>() = rot_mat * normal.head<2>();
+            // Or simply in a single operation:
+            normal << -slope * normInv * rot_mat.toRotationMatrix().col(0), normInv;
+        }
+    }
+
+    HeightmapFunction unidirectionalRandomPerlinGround(
+        double wavelength, std::size_t numOctaves, double orientation, uint32_t seed)
+    {
+        const Eigen::Rotation2D<double> rot_mat(orientation);
+
+        // Set seed of Random Perlin Process
+        PCG32 pcg32_generator = PCG32(seed);
+        RandomPerlinProcess perlinProcess = RandomPerlinProcess(wavelength, numOctaves);
+        perlinProcess.reset(pcg32_generator);
+
+        // Constant for avoiding discontinuity at (0.0, 0.0)
+        const double offset = 100.0;
```

Comment:
What is this ?

Reply:
It's a non-ideal fix. Since the 1d perlin process is defined only on t >0, I needed a way to have it well defined for t<0 as well. Using the absolute value of t doesn't work since you end up with a discontinuity at t=0. So instead I shifted the perlin process by a offset. But it doesn't change the fact that the perlin process remains undefined for t < offset.

Reply:
Ok I will check if it can be refactored to work with negative values.